src/yaml/aslYamlLanguageService.ts

```diff
@@ -0,0 +1,231 @@
+ * SPDX-License-Identifier: MIT
+ */
+
+import * as prettier from 'prettier'
+import {
+    DocumentLanguageSettings,
+    getLanguageService as getLanguageServiceVscode,
+    JSONDocument,
+    JSONSchema,
+    LanguageService,
+    LanguageServiceParams,
+} from 'vscode-json-languageservice'
+import { TextDocument } from 'vscode-languageserver-textdocument'
+import {
+    CompletionList,
+    DocumentSymbol,
+    FormattingOptions,
+    Hover,
+    Position,
+    Range,
+    SymbolInformation,
+    TextEdit
+} from 'vscode-languageserver-types'
+import {
+    parse as parseYAML,
+    SingleYAMLDocument,
+    YAMLDocument,
+} from 'yaml-language-server/out/server/src/languageservice/parser/yamlParser07'
+import { YAMLCompletion } from 'yaml-language-server/out/server/src/languageservice/services/yamlCompletion'
+import { YAMLSchemaService } from 'yaml-language-server/out/server/src/languageservice/services/yamlSchemaService'
+import { matchOffsetToDocument } from 'yaml-language-server/out/server/src/languageservice/utils/arrUtils'
+import doCompleteAsl from '../completion/completeAsl'
+
+export const getLanguageService = function(params: LanguageServiceParams, schema: JSONSchema, aslLanguageService: LanguageService): LanguageService {
+    const builtInParams = {}
+
+    const languageService = getLanguageServiceVscode({
+        ...params,
+        ...builtInParams,
+    })
+
+    const requestServiceMock = async function(uri: string): Promise<string> {
+        return new Promise<string>(c => {
+            c(JSON.stringify(schema))
+        })
+    }
+    const schemaService = new YAMLSchemaService(requestServiceMock, params.workspaceContext)
+    // initialize schema
+    schemaService.registerExternalSchema('yasl', ['*.asl.yaml', '*.asl.yml'], schema)
+    schemaService.getOrAddSchemaHandle('yasl', schema)
+
+    const completer = new YAMLCompletion(schemaService)
+
+    languageService.doValidation = async function(
+        textDocument: TextDocument,
+        jsonDocument: JSONDocument,
+        documentSettings: DocumentLanguageSettings
+    ) {
+        const yamlDocument: YAMLDocument = parseYAML(textDocument.getText())
+        const validationResult: any[] = []
+
+        for (const currentYAMLDoc of yamlDocument.documents) {
+            const validation = await aslLanguageService.doValidation(textDocument, currentYAMLDoc)
+            const syd = (currentYAMLDoc as unknown) as SingleYAMLDocument
+            if (syd.errors.length > 0) {
+                validationResult.push(...syd.errors)
+            }
+            if (syd.warnings.length > 0) {
+                validationResult.push(...syd.warnings)
+            }
+
+            validationResult.push(...validation)
+        }
+
+        return validationResult
+    }
+
+    languageService.doComplete = async function(
+        document: TextDocument,
+        position: Position,
+        doc: JSONDocument
```

Comment:
`doc` parameter is not used anywhere - we can delete it.

Reply:
Removed unused parameters from methods in this module.